internal/hardware/graph/nodes.go

```diff
@@ -84,12 +89,39 @@ func getOSInfo() string {
 	return "Linux" // Fallback
 }
 
+// getMachineID reads the machine ID from /etc/machine-id or /sys/class/dmi/id/product_uuid
+func getMachineID() string {
+	// Try /etc/machine-id first (most common and doesn't require root)
+	if data, err := os.ReadFile("/etc/machine-id"); err == nil {
+		if id := strings.TrimSpace(string(data)); id != "" {
+			return id
+		}
+	}
+
+	// Try /sys/class/dmi/id/product_uuid (requires root but is hardware-based)
+	if data, err := os.ReadFile("/sys/class/dmi/id/product_uuid"); err == nil {
+		if uuid := strings.TrimSpace(string(data)); uuid != "" {
+			return uuid
+		}
+	}
+
+	// Fall back to hostname if neither is available
+	if hostname, err := os.Hostname(); err == nil {
+		return hostname
+	}
```

Comment:
This seems really odd. the id from `/sys/class/dmi/id/product_uuid` is different from `/etc/machine-id`. the latter is an OS specific id. the former is set by the hardware manufacture and encoded directly in the BIOS. So they are separate concepts. 

And then hostname is a network identifer which is completely different.

We should all three of these as tags but only the machine id (fall back to system id) as the name. those should exist on any linux host.